rules/regexp/misc.lua

```diff
@@ -99,4 +99,19 @@ reconf['LEAKED_PASSWORD_SCAM_RE'] = {
   group = 'scams'
 }
 
-rspamd_config:register_dependency('LEAKED_PASSWORD_SCAM', 'BITCOIN_ADDR')
\ No newline at end of file
+rspamd_config:register_dependency('LEAKED_PASSWORD_SCAM', 'BITCOIN_ADDR')
+
+-- Heurististic for detecting InterPlanetary File System (IPFS) gateway URLs:
+-- These contain "ipfs" somewhere (either in the FQDN or the URL path) and a
+-- content identifier (CID), comprising of either "qm", followed by 46 alphanumerical
+-- characters (CIDv0), or a CIDv1 of an alphanumerical string of unspecified length,
+-- depending on the hash algorithm used.
+local ipfs_cid = '/(qm[a-z0-9]{44}|[a-z0-9]{45,256})/{url}i'
```

Comment:
This regexp will be very bad from the [performance](https://intel.github.io/hyperscan/dev-reference/performance.html#bounded-repeats-in-streaming-mode) considerations for Hyperscan (and probably PCRE as well). I'd appreciate if we can use something else but `{45,256}` here.

Reply:
Thank you. I will look into this, and try to come up with a less costly regexp for parsing IPFS CIDv1s.